File: src/main/java/com/moxie/taosha/code_generator_annotation/util/NameConversion.java
package com.moxie.taosha.code_generator_annotation.util;

/**
 * @ClassName NameConversion
 * @Description 名称转换
 * @Author zhangMin
 * @Date 2019/1/17 11:43
 * @Version 1.0
 **/
public class NameConversion {

    public static String humpNameToBigHumpName(String humpName){
        if(Character.isUpperCase(humpName.charAt(0))) {
            return humpName;
        } else {
            return (new StringBuilder()).append(Character.toUpperCase(humpName.charAt(0))).append(humpName.substring(1)).toString();
        }
    }

}

File: src/main/java/com/moxie/taosha/code_generator_kotlin/util/NameConversion.kt
package com.moxie.taosha.code_generator_kotlin.util

/**
 * @ClassName NameConversion
 * @Description 名称改成驼峰
 * @Author zhangMin
 * @Date 2018/9/17 16:56
 * @Version 1.0
 **/

//大写下滑线转大驼峰
fun underLineToBigHumpName(underLineStr:String):String{
    val bigHumpName: StringBuffer = StringBuffer();

    if(underLineStr.length <= 1){
        return underLineStr
    }


    underLineStr.split("_").forEach(){
        bigHumpName.append( it.substring(0, 1).toUpperCase())
        bigHumpName.append(it.substring(1, it.length).toLowerCase())
    }

    return bigHumpName.toString();
}

//大写下滑线转驼峰
fun underLineToHumpName(underLineStr:String):String{
    val bigHumpName: StringBuffer = StringBuffer();

    if(underLineStr.length <= 1){
        return underLineStr
    }

    underLineStr.split("_").forEachIndexed(){index, it ->
        if(index == 0){
            bigHumpName.append( it.substring(0, 1).toLowerCase())
        }else{
            bigHumpName.append( it.substring(0, 1).toUpperCase())
        }

        bigHumpName.append(it.substring(1, it.length).toLowerCase())
    }

    return bigHumpName.toString();
}